controllers/nnf_workflow_controller_helpers.go

```diff
@@ -1057,106 +1058,50 @@ func (r *NnfWorkflowReconciler) createOrUpdateContainerJobsIfNecessary(ctx conte
-					{
-						Effect:   "NoSchedule",
-						Key:      "cray.nnf.node",
-						Operator: "Equal",
-						Value:    "true",
-					},
-				}
-
-				// TODO Volumes
-				// podSpec.Volumes
-				// ssh is necessary for mpi
-				// setupSSHAuthVolumes(manager, podSpec)
-				// setupLustreVolumes(ctx, manager, podSpec, filesystems.Items)
-				// TODO: add Volumes to DS, append if customer specifies; check for conflicts
-				// vol := corev1.Volume{
-				// 	Name: "foo-local-storage",
-				// 	VolumeSource: corev1.VolumeSource{
-				// 		HostPath: &corev1.HostPathVolumeSource{
-				// 			Path: "/path/to/thing"}}}
-				// podTemplateSpec.Spec.Volumes = append(podTemplateSpec.Spec.Volumes, vol)
-				// TODO: get mounts from workflow
-			}
-
-			// This is a timeout for the job, if set, when the deadline hits, the job will kill all
-			// pods and fail the job.
-			if profile.Data.ActiveDeadlineSeconds > 0 {
-				job.Spec.ActiveDeadlineSeconds = &profile.Data.ActiveDeadlineSeconds
-			}
+		// Get the volumes to mount into the containers
+		volumes, err := r.getContainerVolumes(ctx, workflow, dwArgs, nnfNode)
+		if err != nil {
+			return nnfv1alpha1.NewWorkflowErrorf("could not determine the list of volumes need to create container job %s:", job.Name).WithError(err).WithFatal()
+		}
 
-			// This defaults to 6 and is the maximum number of pod started before considering the
-			// job failed. I don't believe this can be turned off.
-			job.Spec.BackoffLimit = &profile.Data.BackoffLimit
+		// Set the job's podspec from the container profile and add in volumes
+		r.setContainerJobPodSpec(ctx, workflow.Name, job, *profile, nnfNode, volumes)
 
-			return nil
+		// This is a timeout for the job, if set, when the deadline hits, the job will kill all
+		// pods and fail the job.
+		if profile.Data.ActiveDeadlineSeconds > 0 {
+			job.Spec.ActiveDeadlineSeconds = &profile.Data.ActiveDeadlineSeconds
 		}
+		// This defaults to 6 and is the maximum number of pod started before considering the
+		// job failed. I don't believe this can be turned off.
+		job.Spec.BackoffLimit = &profile.Data.BackoffLimit
 
-		result, err := ctrl.CreateOrUpdate(ctx, r.Client, job, mutateFn)
+		err = r.Create(ctx, job)
 		if err != nil {
-			return err
-		}
-
-		if result == controllerutil.OperationResultCreated {
-			log.Info("Created Job", "object", client.ObjectKeyFromObject(job).String())
-		} else if result == controllerutil.OperationResultUpdated {
-			log.Info("Updated Job", "object", client.ObjectKeyFromObject(job).String())
+			if !apierrors.IsAlreadyExists(err) {
+				return err
+			}
+			log.Info("Job already exists", "object", client.ObjectKeyFromObject(job).String())
 		}
+		log.Info("Created Job", "object", client.ObjectKeyFromObject(job).String())
 	}
 
 	return nil
 }
 
+// Retrieve the computes for the workflow and find their local nnf nodes
 func (r *NnfWorkflowReconciler) getNnfNodesForContainers(ctx context.Context, workflow *dwsv1alpha1.Workflow) ([]string, error) {
```

Comment:
I surprised that containers is the only area that we need to know the list of Rabbits associated with the compute nodes. Just thinking this could named, getNnfNodesForComputes.

Reply:
Good call. I renamed it.